spec/api/abuse_spec.rb

```diff
@@ -1,170 +1,136 @@
 require 'spec_helper'
 
 def create_comment_flag(comment_id, user_id)
-  create_flag("/api/v1/comments/" + comment_id + "/abuse_flag", user_id)
+  put "/api/v1/comments/#{comment_id}/abuse_flag", user_id: user_id
 end
 
 def create_thread_flag(thread_id, user_id)
-  create_flag("/api/v1/threads/" + thread_id + "/abuse_flag", user_id)
+  put "/api/v1/threads/#{thread_id}/abuse_flag", user_id: user_id
 end
 
 def remove_thread_flag(thread_id, user_id)
-  remove_flag("/api/v1/threads/" + thread_id + "/abuse_unflag", user_id)
+  put "/api/v1/threads/#{thread_id}/abuse_unflag", user_id: user_id
 end
 
 def remove_comment_flag(comment_id, user_id)
-  remove_flag("/api/v1/comments/" + comment_id + "/abuse_unflag", user_id)
+  put "/api/v1/comments/#{comment_id}/abuse_unflag", user_id: user_id
 end
 
-def create_flag(put_command, user_id)
-  if user_id.nil?
-    put put_command
-  else
-    put put_command, user_id: user_id
+describe 'Abuse API' do
+  before(:each) { set_api_key_header }
+  let(:thread) { create_comment_thread_and_comments }
+  let(:comment) { thread.comments.first }
+  let(:user) { thread.author }
+
+  describe 'flag a comment as abusive' do
+    it 'create or update the abuse_flags on the comment' do
+      prev_abuse_flaggers_count = comment.abuse_flaggers.length
+      create_comment_flag(comment.id, user.id)
+
+      comment.reload
+      comment.abuse_flaggers.count.should == prev_abuse_flaggers_count + 1
```

Comment:
Prefer `expect` syntax.  Also there are more readable matchers we can use here.
https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers

`expect(comment.abuse_flaggers).to have(prev_abuse_flaggers_count + 1).items`